data_registry/process_manager/task/coverage.py

```diff
@@ -0,0 +1,70 @@
+import gzip
+import shutil
+import tempfile
+from pathlib import Path
+
+import ocdscardinal
+
+from data_registry import models
+from data_registry.process_manager.util import TaskManager, skip_if_not_started
+from exporter.util import Export
+
+
+def get_keys_for_building_block(coverage, building_block):
+    return [key for key in coverage if key.endswith(building_block)]
+
+
+class Coverage(TaskManager):
+    final_output = False
+    json_file_name = "full.jsonl"
+
+    def get_export(self):
+        return Export(self.job.pk, f"{self.json_file_name}.gz")
+
+    def run(self):
+        export = self.get_export()
+        export.lock()
+        json_file_path = export.path
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tmpdir = Path(tmpdirname)
+            infile = tmpdir / self.json_file_name
+            with gzip.open(json_file_path) as i, infile.open("wb") as o:
+                shutil.copyfileobj(i, o)
+                coverage = ocdscardinal.coverage(o.name)
```

Comment:
Unindent after the shutil.copyfileobj, so that the file is flushed to disk and closed, before reading with ocdscardinal. You'll need to change `o.name` to `str(infile)`.